Source/ACE/Command/Handlers/DebugCommands.cs

```diff
@@ -65,17 +65,19 @@ public static void HandleDebugTeleportXYZ(Session session, params string[] param
         [CommandHandler("grantxp", AccessLevel.Developer, CommandHandlerFlag.RequiresWorld, 1)]
         public static void HandleGrantXp(Session session, params string[] parameters)
         {
-            ulong xp = 0;
-
-            if (parameters?.Length > 0 && ulong.TryParse(parameters[0], out xp))
-            {
-                session.Player.GrantXp(xp);
-            }
-            else
+            if (parameters?.Length > 0)
             {
-                ChatPacket.SendServerMessage(session, "Usage: /grantxp 1234", ChatMessageType.Broadcast);
-                return;
+                ulong xp = 0;
+                string xpAmountToParse = parameters[0].Length > 12 ? parameters[0].Substring(0, 12) : parameters[0];

```

Comment:
If we do not limit the experience points to the maximum, the client will not lock to the "Infinity!" text. TotalExperience should NOT exceed: 191,226,310,247